packages/cli/test/unit/commands/ps/index.unit.test.ts

```diff
@@ -0,0 +1,279 @@
+import {stdout, stderr} from 'stdout-stderr'
+import Cmd  from '../../../../src/commands/ps'
+import runCommand from '../../../helpers/runCommand'
+const nock = require('nock')
+const {expect} = require('chai')
+const strftime = require('strftime')
+const hourAgo = new Date(Date.now() - (60 * 60 * 1000))
+const hourAgoStr = strftime('%Y/%m/%d %H:%M:%S %z', hourAgo)
+function stubAccountQuota(code: number, body: Record<string, string | number | [] | [Record<string, string | number>]>) {
+  nock('https://api.heroku.com:443')
+    .get('/account/features/free-2016')
+    .reply(200, {enabled: true})
+  nock('https://api.heroku.com:443')
+    .get('/apps/myapp/dynos')
+    .reply(200, [{command: 'bash', size: 'Eco', name: 'run.1', type: 'run', updated_at: hourAgo, state: 'up'}])
+  nock('https://api.heroku.com:443', {
+    reqHeaders: {Accept: 'application/vnd.heroku+json; version=3.process_tier'},
+  })
+    .get('/apps/myapp')
+    .reply(200, {process_tier: 'eco', owner: {id: '1234'}, id: '6789'})
+  nock('https://api.heroku.com:443')
+    .get('/account')
+    .reply(200, {id: '1234'})
+  nock('https://api.heroku.com:443', {
+    reqHeaders: {Accept: 'application/vnd.heroku+json; version=3.account-quotas'},
+  })
+    .get('/accounts/1234/actions/get-quota')
+    .reply(code, body)
+}
+
+function stubAppAndAccount() {
+  nock('https://api.heroku.com:443', {
+    reqHeaders: {Accept: 'application/vnd.heroku+json; version=3.process_tier'},
+  })
+    .get('/apps/myapp')
+    .reply(200, {process_tier: 'basic', owner: {id: '1234'}})
+  nock('https://api.heroku.com:443')
+    .get('/account')
+    .reply(200, {id: '1234'})
+}
+
+describe('ps', () => {
+  beforeEach(() => {
+    nock.cleanAll()
+    stdout.start()
+    stderr.start()
+  })
+  it('shows dyno list', function () {
+    const api = nock('https://api.heroku.com:443')
+      .get('/apps/myapp/dynos')
+      .reply(200, [
+        {command: 'npm start', size: 'Eco', name: 'web.1', type: 'web', updated_at: hourAgo, state: 'up'}, {command: 'bash', size: 'Eco', name: 'run.1', type: 'run', updated_at: hourAgo, state: 'up'},
+      ])
+    stubAppAndAccount()
+    return runCommand(Cmd, [
+      '--app',
+      'myapp',
+    ])
```

Comment:
Old test setup that needs to be migrated to [this format](https://github.com/heroku/cli/pull/2589/commits/3ee6844aca058d321549463a56fa2edcca8c579b#r1453786810)